File: KCD/HMPlanToShowsBoldFontTransformer.m
//
//  HMPlanToShowsBoldFontTransformer.m
//  KCD
//
//  Created by Hori,Masaki on 2014/08/29.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMPlanToShowsBoldFontTransformer.h"

@implementation HMPlanToShowsBoldFontTransformer
+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[NSValueTransformer setValueTransformer:[self new] forName:@"HMPlanToShowsBoldFontTransformer"];
	});
}

+ (Class)transformedValueClass
{
	return [NSNumber class];
}
+ (BOOL)allowsReverseTransformation
{
	return NO;
}

- (id)transformedValue:(id)value
{
	if(![value isKindOfClass:[NSNumber class]]) return @NO;
	
	NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
	if(![ud boolForKey:@"showsPlanColor"]) return @NO;
	
	if([value integerValue] == 0) return @NO;
	
	return @YES;
}
@end

File: KCD/HMPlanToShowsBoldFontTransformer.swift
//
//  HMPlanToShowsBoldFontTransformer.swift
//  KCD
//
//  Created by Hori,Masaki on 2014/12/21.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMPlanToShowsBoldFontTransformer: NSValueTransformer
{
	override class func load() {
		NSValueTransformer.setValueTransformer(HMPlanToShowsBoldFontTransformer(), forName: "HMPlanToShowsBoldFontTransformer")
	}
	
	override class func transformedValueClass() -> AnyClass {
		return NSNumber.self
	}
	
	override class func allowsWeakReference() -> Bool { return false }
	
	override func transformedValue(value: AnyObject?) -> AnyObject? {
		let numValue = value as? NSNumber
		if numValue == nil {
			return false
		}
		
		if !HMUserDefaults.hmStandardDefauls().showsPlanColor {
			return false
		}
		if numValue == 0 {
			return false
		}
		return true
	}
	
}
